### PR TITLE
Increase history capacity by 1

### LINKED ISSUE #175: [PE-D][Tester A] Undo history allows only 9 undos despite UG stating up to 10
Steps to reproduce:
1. Add 11 internships (add A /t r, add B /t r, ... x11)
2. Attempt to undo 11 times

Expected: 10 successful undos before "no more undo history" error per UG documentation.

Actual: Only 9 successful undos before "Error: there is no more undo history."
The 11th add (outside undo window) is expected, but the 10th undo also fails unexpectedly.

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by the-RoaringCat**
close #175
Increase by `1` to include the current version since it is also saved in the undo stack.

### CODE REVIEW

File: src/main/java/seedu/goldencompass/operation/OperationHistory.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 public class OperationHistory {
     private static final int CAPACITY = 10;
-
+    private static final int REAL_CAPACITY = CAPACITY + 1; //since current version also takes up a space in stack.
```
Comment (liny666) [inline src/main/java/seedu/goldencompass/operation/OperationHistory.java:8]: Introducing `REAL_CAPACITY` implies `CAPACITY` is somehow fake. Consider keeping the check inline instead:
```java
if (undoStack.size() == CAPACITY + 1) { // +1 for current state
    undoStack.removeLast();
}
```
This avoids a second constant and makes the intent clear at the call site.

**Review by liny666 — COMMENTED** [review]
No regression test for the capacity fix — consider adding a test that performs 12+ commands and verifies exactly 10 undos succeed while the 11th returns null. Given this was a silent off-by-one bug (#175), a test would prevent it from regressing.

File: src/main/java/seedu/goldencompass/operation/OperationHistory.java [hunk 2]
```diff
@@ -16,7 +16,7 @@ public OperationHistory() {
     }
 
     public void saveSnapshot(OperationSnapshot snapshot) {
-        if(undoStack.size() == CAPACITY) {
+        if(undoStack.size() == REAL_CAPACITY) {
```
Comment (liny666) [inline src/main/java/seedu/goldencompass/operation/OperationHistory.java:19]: Minor: missing space before `(` — should be `if (undoStack.size() ...`. Pre-existing, but worth fixing while touching this line.
